app/assets/javascripts/application.js

```diff
@@ -1,4 +1,157 @@
 $(document).ready(function(){
 
+  function Task(params){
+    if(params) this.data = params;
+    this.render();
+  }
 
-});
+
+  function List(params, tasksHtml){
+    if(params){
+      this.data = params;
+      this.tasksHtml = tasksHtml;
+      this.render();
+    }
+  }
+
+  List.fetch = function(){
+    var request = $.getJSON("http://localhost:3000/lists")
+    .then(function(response){
+      var lists = [];
+      for(var i = 0; i < response.length; i++){
+        lists.push(new List(response[i]))
+      }
+      return lists
+    })
+    .fail(function(response){
+      console.log("lists fetch fail")
+    })
+    return request
+  }
+
+  List.prototype = {
+    var lists;
+    var tasks;
+    var params = {title: $(this).siblings("[name=title]").val()}
+    render: function(){
+      this.view = "<div class='list'><input type='hidden' name='id' value='" + (this.data.id || "") + "' /><input type='text' name='title' value='" + (this.data.title || "") + "' />";
+      if(!this.data.id){
+        this.view += "<button type='button' class='create'>Create</button>";
+      }else{
+        this.view += "<button type='button' class='save'>Save</button><button type='button' class='delete'>Delete</button>";
+        this.view += "<ul>" + this.tasksHtml + "</ul>";
+      }
+      this.view += "</div>";
+    }
+  },
+
+  Tasks.fetch = function(){
+      var url = "http://localhost:3000/lists" + this.id + "/tasks";
```

Comment:
remove the protocol, host, and port here to make this application easier to deploy